app/client/src/reducers/uiReducers/mainCanvasReducer.ts

```diff
@@ -0,0 +1,45 @@
+import { createImmerReducer } from "utils/AppsmithUtils";
+import {
+  ReduxAction,
+  ReduxActionTypes,
+  UpdateCanvasPayload,
+} from "constants/ReduxActionConstants";
+import { MAIN_CONTAINER_WIDGET_ID } from "constants/WidgetConstants";
+import { UpdateCanvasLayoutPayload } from "actions/controlActions";
+
+const initialState: MainCanvasReduxState = {
+  width: 0,
+  height: 0,
+};
+
+const mainCanvasReducer = createImmerReducer(initialState, {
+  [ReduxActionTypes.INIT_CANVAS_LAYOUT]: (
+    state: MainCanvasReduxState,
+    action: ReduxAction<UpdateCanvasPayload>,
+  ) => {
+    const mainCanvas =
+      action.payload.widgets &&
+      action.payload.widgets[MAIN_CONTAINER_WIDGET_ID];
+
+    return {
```

Comment:
Since we're using immer, we can do this as so:
```suggestion
    state.width = mainCanvas?.rightColumn || state.width;
    state.height = mainCanvas?.minHeight || state.height;
```